### PR TITLE
Fix import s3

### PR DESCRIPTION
**Pull request by hmeiland**
renewing pull request for fixing import of s3 backed cvmfs (closing old one due to merge conflicts)

### CODE REVIEW

**Comment by hmeiland** [comment]
test with azurite not possible at this moment: azurite crashes on import, while azure blob handles correctly.

**Comment by jblomer** [comment]
Many thanks! Do you know why azurite crashes?

**Comment by hmeiland** [comment]
had to merge azureblob into this one to get all the changes working; now the azurite works for import as well...
need to discuss on removal of assert in signing_tool...

**Comment by hmeiland** [comment]
> Many thanks! Do you know why azurite crashes?

yes -> was not pointing to the correct tcp port no... ;( ,fixed now

File: cvmfs/server/cvmfs_server_import.sh [hunk 1]
```diff
@@ -168,19 +170,21 @@ cvmfs_server_import() {
...
-  local i=0
-  while [ $i -lt 256 ]; do
-    needed_items="$needed_items ${storage_location}/data/$(printf "%02x" $i)"
-    i=$(($i+1))
-  done
-  for item in $needed_items; do
-    [ -e $item ] || die "$item missing"
-    [ $chown_backend -ne 0 ] || [ x"$cvmfs_user" = x"$(stat -c%U $item)" ] || die "$item not owned by $cvmfs_user"
-  done
+  if [ x"$(is_local_upstream $upstream)" != "x" ]; then
```
Comment (jblomer) [inline cvmfs/server/cvmfs_server_import.sh]:
```suggestion
  if is_local_upstream $upstream; then
```

**Review by jblomer — CHANGES_REQUESTED** [review]
Many thanks!  Import with S3 would be a great new functionality!  Let's merge #2776 first and rebase this on on top. On the reflog checksum, are you sure that removing the assert is necessary, i.e. does importing break otherwise, and where?

**Comment by hmeiland** [comment]
> Many thanks! Import with S3 would be a great new functionality! Let's merge #2776 first and rebase this on on top. On the reflog checksum, are you sure that removing the assert is necessary, i.e. does importing break otherwise, and where?

Yes, the assert is clearly the one breaking the import; I can not find another way to satisfy the assert condition; If you feel there should be, let's set up a short call to discuss...

**Comment by jblomer** [comment]
@cernvm-bot help

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Comment by jblomer** [comment]
@cernvm-bot unittest

**Comment by jblomer** [comment]
@cernvm-bot cpplint

**Comment by jblomer** [comment]
@cernvm-bot tidy

**Comment by jblomer** [comment]
The change in the reflog checksum handling looks fine to me. The PR is replaced by #2795 which contains some additional fixes.

**Pull request closed without merging**